neo4j-jdbc/src/main/java/org/neo4j/jdbc/Neo4jDataSource.java

```diff
@@ -68,4 +83,57 @@
 	@Override public boolean isWrapperFor(Class<?> iface) throws SQLException {
 		return Wrapper.isWrapperFor(iface, this.getClass());
 	}
+
+	public String getServerName() {
+		return serverName;
+	}
+
+	public void setServerName(String serverName) {
+		this.serverName = serverName;
+	}
+
+	public String getUser() {
+		return user;
+	}
+
+	public void setUser(String user) {
+		this.user = user;
+	}
+
+	public String getPassword() {
+		return password;
+	}
+
+	public void setPassword(String password) {
+		this.password = password;
+	}
+
+	public int getPortNumber() {
+		return portNumber;
+	}
+
+	public void setPortNumber(int portNumber) {
+		this.portNumber = portNumber;
+	}
+
+	public String getUrl(String protocol) {
+		String url = NEO4J_JDBC_PREFIX + protocol + "://" + getServerName() + ((getPortNumber() > 0) ? ":" + getPortNumber() : "") + "?" + ((!getIsSsl()) ?
+				"nossl," :
+				"");
+		if (Objects.nonNull(getUser())) {
+			url += "user=" + getUser();
+			if (Objects.nonNull(getPassword())) {
```

Comment:
should we really output passwords?

Reply:
how do other datasources do this?

Reply:
https://github.com/pgjdbc/pgjdbc/blob/43e6505e3aa16e6acdf08f02f2dd1e3cf131ac3e/pgjdbc/src/main/java/org/postgresql/ds/common/BaseDataSource.java#L183

This is PostgresSQL JDBC driver, it has both getPassword and getUrl as public methods.